File: TDDMicroExercises.ProposedSolution/CSharp/TelemetrySystem/TelemetryClientConnection.cs
using System;

namespace TDDMicroExercises.OneSolution.TelemetrySystem
{
    public class TelemetryClientConnection : IConnection
    {
        private bool _onlineStatus;
        private readonly Random _connectionEventsSimulator = new Random(42);

        public bool OnlineStatus
        {
            get { return _onlineStatus; }
        }

        public void Connect(string telemetryServerConnectionString)
        {
            if (string.IsNullOrEmpty(telemetryServerConnectionString))
            {
                throw new ArgumentNullException();
            }

            // simulate the operation on a real modem
            bool success = _connectionEventsSimulator.Next(1, 10) <= 8;

            _onlineStatus = success;
        }

        public void Disconnect()
        {
            _onlineStatus = false;
        }
    }
}

File: TDDMicroExercises.ProposedSolution/Javascript/telemetry-system/telemetry-client-connection.js
TelemetryClientConnection = function() {

	this._onlineStatus = false;
	
};



TelemetryClientConnection.prototype = {

	// simulate the operation on a real modem
	_connectionEventsSimulator: function (min, max) {
		var delta = max + 1 - min;
		return min + Math.floor(delta * Math.random());
	},

	onlineStatus: function () {
		return this._onlineStatus;
	},
	
	connect: function(telemetryServerConnectionString) {
			if (typeof(telemetryServerConnectionString) === 'undefined' || telemetryServerConnectionString === '') {
				throw 'missing telemetryServerConnectionString parameter';
			}

			// simulate the operation on a real modem
			var success = this._connectionEventsSimulator(1, 10) <= 8;

			this._onlineStatus = success;
	},
	
	disconnect: function () {
		this._onlineStatus = false;
	}

};
